src/ray/core_worker/core_worker.cc

```diff
@@ -2886,19 +2886,23 @@ void CoreWorker::HandleUpdateObjectLocationBatch(
     return;
   }
   const auto &node_id = NodeID::FromBinary(request.node_id());
-  const auto &object_location_states = request.object_location_states();
+  const auto &object_location_updates = request.object_location_updates();
 
-  for (const auto &object_location_state : object_location_states) {
-    const auto &object_id = ObjectID::FromBinary(object_location_state.object_id());
-    const auto &state = object_location_state.state();
+  for (const auto &object_location_update : object_location_updates) {
+    const auto &object_id = ObjectID::FromBinary(object_location_update.object_id());
 
-    if (state == rpc::ObjectLocationState::ADDED) {
+    if (object_location_update.has_spilled_url()) {
+      AddSpilledObjectLocationOwner(
+          object_id,
+          object_location_update.spilled_url(),
+          NodeID::FromBinary(object_location_update.spilled_node_id()));
+    }
+
+    if (object_location_update.has_in_plasma() && object_location_update.in_plasma()) {
```

Comment:
Why don't we just keep ADDED / REMOVED here? Feel like `object_location_update.has_in_plasma() && object_location_update.in_plasma()` is a bit hard to understand